crates/tabby-common/src/config.rs

```diff
@@ -325,30 +331,37 @@ pub struct AnswerConfig {
 pub struct CodeRepository {
     pub git_url: String,
     pub source_id: String,
+    pub dir: String,
 }
 
 impl CodeRepository {
     pub fn new(git_url: &str, source_id: &str) -> Self {
+        let dir = if RepositoryConfig::resolve_is_local_dir(git_url) {
```

Comment:
what happens if dir doesn't contain valid string? e.g `git:12a5h`, `provided_repository:12a5h`? Have you tested these cases (configure repository from webserver)?

Reply:
I have add some test for these case.